add towny join and leave event

MadServlet opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/DenizenScript/Depenizen?pullRequest=395) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/DenizenScript/Depenizen?pullRequest=395) before we can accept your contribution.<br/><hr/>**james.castillo.jc@outlook.com** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/DenizenScript/Depenizen?pullRequest=395) it.</sub>

MadServlet commented:
might need some help with this one.
I think the revert I did didn't apply at all

mcmonkey4eva commented on src/main/java/com/denizenscript/depenizen/bukkit/events/towny/PlayerJoinsTownScriptEvent.java:
```diff
@@ -0,0 +1,58 @@
+    // @Triggers when a player joins a Towny town.
+    //
+    // @Context
+    // <context.town> Returns town the player is joining in.
```
English grammar: should be `returns the town` - lowercase `r` and include `the`
also remove `in` from the end.

mcmonkey4eva commented on src/main/java/com/denizenscript/depenizen/bukkit/events/towny/PlayerJoinsTownScriptEvent.java:
```diff
@@ -0,0 +1,58 @@
+    public ObjectTag getContext(String name) {
+        switch (name) {
+            case "town":
+                return new TownTag(event.getTown());
```
Should be one line, `case "town": return new ...`

mcmonkey4eva commented on src/main/java/com/denizenscript/depenizen/bukkit/events/towny/PlayerJoinsTownScriptEvent.java:
```diff
@@ -0,0 +1,58 @@
+        this.event = event;
+        fire(event);
+    }
+
```
redundant newline

mcmonkey4eva commented on src/main/java/com/denizenscript/depenizen/bukkit/events/towny/PlayerJoinsTownScriptEvent.java:
```diff
@@ -0,0 +1,58 @@
+    // @Events
+    // towny player joins town
+    //
+    // @Cancellable false
```
Don't do this, just remove the line

MadServlet commented:
requested changes applied.

-corrected grammatical error in documentation
-changed switch-case to one line
-removed redundant new lines
-added space after `-->` documentation
-removed explicit cancellable in documentation

mcmonkey4eva commented:
Please do not touch the "resolve" button
